pylint/config/config_file_parser.py

```diff
@@ -79,7 +79,8 @@ def _parse_toml_file(self, file_path: Path) -> tuple[dict[str, str], list[str]]:
                 for config, value in values.items():
                     value = _parse_rich_type_value(value)
                     config_content[config] = value
-                    options += [f"--{config}", value]
+                    if value.lower() != "false":
```

Comment:
I think we should also check the type of the config option. If somebody put in `false` for a non-`store_true` option we don't want to this this. A test for this would also be good!

Reply:
Sounds good! Do you know if there's an efficient way to check the type of config? I was looking at the function "_convert_option_to_argument" in pylint/config/utils.py, but not sure if that would work.

Reply:
No, although you could perhaps wait with parsing them until you reach:

```python
https://github.com/pylint-dev/pylint/blob/4cdf9222ae7ed5a3122f633d04b8d8c144e60c1c/pylint/config/arguments_manager.py#L211-L214
```

This should have knowledge of all registered arguments.